src/main.rs

```diff
@@ -331,6 +342,13 @@ fn merge_avgs(
             if a.status > avg_status {
                 avg_status = a.status;
             }
+
+            if !a.avg_type.is_empty() {
+                if !avg_type.is_empty() {
+                    avg_type += " ";
+                }
+                avg_type += &a.avg_type;
```

Comment:
It would be much better if you remove any duplicated type here instead of a simple concatenation.

Reply:
Used `HashSet` to remove duplicates.